gitlab/v4/objects/merge_requests.py

```diff
@@ -409,6 +409,30 @@ def merge(
         self._update_attrs(server_data)
         return server_data
 
+    @cli.register_custom_action("ProjectMergeRequest")
+    @exc.on_http_error(exc.GitlabGetError)
+    def get_reviewers(self, **kwargs: Any) -> Dict[str, Any]:
+        """List the reviewers.
+
+        Args:
+            per_page: Number of items to retrieve per request
+            page: ID of the page to return (starts with page 1)
+            **kwargs: Extra options to send to the server (e.g. sudo)
+
+        Raises:
+            GitlabAuthenticationError: If authentication is not correct
+            GitlabGetError: If the list could not be retrieved
+
+        Returns:
+            The list of reviewers
+        """
+
+        path = f"{self.manager.path}/{self.encoded_id}/reviewers"
+        result = self.manager.gitlab.http_get(path, **kwargs)
+        if TYPE_CHECKING:
+            assert not isinstance(result, requests.Response)
+        return result
```

Comment:
Thanks a lot for your contribution @kesmik! 

The reviewers endpoint presumably returns a paginated response (though I admit we'll rarely see more than 20 reviewers, but still). This means we should use `http_list` for this endpoint.

We usually don't write custom `get_*` methods for this, but rather add a manager uses the `ListMixin` as a base. Then people can call `mr.reviewers.list()`, and we have a few other convenience attributes there to control list filters etc. I admit the MR manager violates this with a few custom methods here already.

A recent example of a similar addition was for MR draft notes which you can also list:
https://github.com/python-gitlab/python-gitlab/pull/2728/files

Would you be able to rework your PR to use the manager instead @kesmik? Thanks a lot!

Reply:
Hello, thanks for feedback I will check and try to rewrite as you recommending, it seems better way than mine. Initially I thought its not possible since projects already has mergerequest atttibute ;D

Reply:
Ok so I have refactored and moved api call to reviewers manager. I still have a problem for "reviewers" attribute naming.
merge request api (https://docs.gitlab.com/ee/api/merge_requests.html#get-single-mr) returns reviewers in JSON:
```json
"reviewers": [{
    "name": "Miss Monserrate Beier",
    "username": "axel.block",
    "id": 12,
    "state": "active",
    "avatar_url": "http://www.gravatar.com/avatar/46f6f7dc858ada7be1853f7fb96e81da?s=80&d=identicon",
    "web_url": "https://gitlab.example.com/axel.block"
  }],
```

That leads that I can not have manager named "reviewers",  because it will override original reviewers attribute which is supposed to be list.

Also I can not somehow override it with a response of ../reviewers, because format differs..
https://docs.gitlab.com/ee/api/merge_requests.html#get-single-merge-request-reviewers
```json
[
  {
    "user": {
      "id": 1,
      "name": "John Doe1",
      "username": "user1",
      "state": "active",
      "avatar_url": "http://www.gravatar.com/avatar/c922747a93b40d1ea88262bf1aebee62?s=80&d=identicon",
      "web_url": "http://localhost/user1"
    },
    "state": "unreviewed",
    "created_at": "2022-07-27T17:03:27.684Z"
  }
```

Is there anythong I can solve in order to achieve mr.reviewers.list() idea so that:
mr = project.mergerequests.get(1)
reviewers = mr.reviewers.list()
would satisfy:
assert mr.reviewers[0]["name"] == reviewers[0]["user"]["name"]

I think it is not possible with python, and I still need to call that manager somehow diffrently? 

Any help would be appreciated, I would be able to clean this PR

Reply:
@kesmik that's a tricky one yes. We have something similar in https://github.com/python-gitlab/python-gitlab/blob/main/gitlab/mixins.py#L749-L761, but since the responses between this endpoint and the attribute are not consistent, I'd say it might make sense to rename the manager attribute.

`mr_reviewers` looks a bit off to me because of the abbreviation, I'd rather expand that to `merge_request_reviewers` in that case. But since we don't actually get just a list of users but also other attributes, I like your other suggestions more.. How about something like `mr.reviewer_details.list()`? Then if GitLab expands the endpoint with other attributes, we're still accurately describing what's being fetched.

But it's definitely a weird endpoint. The `/participants` endpoint is right there as well and does not nest the user details.

Reply:
Yep it is really weird for me as well. So I have rebased all these changes and presuming it is ready to be merged. If anything is needed to be changed let me know I will try my best!

Reply:
Thanks a lot for the rework, just went through this one last time and just had a small note that I forgot, based on our previous rework, let me know what you think and then we should be good to go!